cate/ds/esa_cci_odp.py

```diff
@@ -446,81 +464,61 @@ def sync(self,
              time_range: Tuple[datetime, datetime]=None,
              protocol: str=None,
              monitor: Monitor=Monitor.NONE) -> Tuple[int, int]:
-        selected_file_list = self._find_files(time_range)
-        if not selected_file_list:
-            return 0, 0
-
-        if protocol is None:
-            protocol = _ODP_PROTOCOL_HTTP
 
         if protocol == _ODP_PROTOCOL_HTTP:
-            dataset_dir = self.local_dataset_dir()
-
-            # Find outdated files
-            outdated_file_list = []
-            for file_rec in selected_file_list:
-                filename, _, _, file_size, url = file_rec
-                dataset_file = os.path.join(dataset_dir, filename)
-                # todo (forman, 20160915): must perform better checks on dataset_file if it is...
-                # ... outdated or incomplete or corrupted.
-                # JSON also includes "checksum" and "checksum_type" fields.
-                if not os.path.isfile(dataset_file) or (file_size and os.path.getsize(dataset_file) != file_size):
-                    outdated_file_list.append(file_rec)
-
-            if not outdated_file_list:
-                # No sync needed
-                return 0, len(selected_file_list)
-
-            with monitor.starting('Sync ' + self.name, len(outdated_file_list)):
-                bytes_to_download = sum([file_rec[3] for file_rec in outdated_file_list])
-                dl_stat = _DownloadStatistics(bytes_to_download)
-
-                file_number = 1
-                dataset_dir = self.local_dataset_dir()
-                for filename, _, _, file_size, url in outdated_file_list:
-                    if monitor.is_cancelled():
-                        raise InterruptedError
-                    dataset_file = os.path.join(dataset_dir, filename)
-                    sub_monitor = monitor.child(work=1.0)
-
-                    # noinspection PyUnusedLocal
-                    def reporthook(block_number, read_size, total_file_size):
-                        dl_stat.handle_chunk(read_size)
-                        if monitor.is_cancelled():
-                            raise InterruptedError
-                        sub_monitor.progress(work=read_size, msg=str(dl_stat))
-
-                    sub_monitor_msg = "file %d of %d" % (file_number, len(outdated_file_list))
-                    with sub_monitor.starting(sub_monitor_msg, file_size):
-                        urllib.request.urlretrieve(url[protocol], filename=dataset_file, reporthook=reporthook)
-                    file_number += 1
-
-            return len(outdated_file_list), len(selected_file_list)
+            self.make_local(self._master_id(), None, time_range, None, None, monitor)
         else:
-            return 0, 0
+            raise ValueError('Unsupported protocol', protocol)
+        return 0, 0
+
+    def update_local(self,
+                     local_id: str,
+                     time_range: Tuple[datetime, datetime],
```

Comment:
Same here, maybe it's better to do:
```python
time_range: TimeRangeLike.TYPE

time_range = TimeRangeLike.convert(time_range)
```
Then you can pass in other things that can be converted to a datetime Tuple into the method. Of course, if this method does not face user interfaces, this might be redundant.